package-manifest.toml

```diff
@@ -89,10 +89,37 @@ setup_hint = "Run `./tools/ci_download_cockroachdb` to download the necessary bi
 
 [package.internal-dns]
 service_name = "internal_dns"
+source.type = "composite"
+source.packages = [ "dns-server.tar.gz", "internal-dns-customizations.tar.gz" ]
+output.type = "zone"
+
+[package.external-dns]
+service_name = "external_dns"
+source.type = "composite"
+source.packages = [ "dns-server.tar.gz", "external-dns-customizations.tar.gz" ]
+output.type = "zone"
```

Comment:
At a glance, I think this packaging makes sense. I do want to check-in with you though - does this feel weird / unwieldly from your point-of-view?

(For example, I suppose I could support adding `paths` directly to a `composite` type package, so the intermediate "customizations" package wouldn't be needed?)

Reply:
It feels okay.  (Truthfully, this model was a bit unintuitive to me.  I was surprised that if an Omicron package uses a Rust package, it must have the same name as the Rust package.  That made it seem like they're 1-1 with Rust packages.  But they're not, since there are both Omicron packages with no Rust package and Rust packages with no Omicron package.  So at first I thought I was going to have to extend the packaging system if I wanted to use the same Rust package in two different Omicron packages.  I was glad it turned out to be a lot simpler than that, and it's pretty straightforward.  I think I'd need more experience with this to suggest any concrete improvements.  So that's why I say it feels okay.)